graylog2-server/src/main/java/org/graylog2/indexer/datanode/RemoteReindexingMigrationAdapter.java

```diff
@@ -20,14 +20,15 @@
 import org.graylog2.indexer.migration.IndexerConnectionCheckResult;
 import org.graylog2.indexer.migration.RemoteReindexMigration;
 
+import java.net.MalformedURLException;
 import java.net.URI;
 
 public interface RemoteReindexingMigrationAdapter {
     enum Status {
         NOT_STARTED, STARTING, RUNNING, ERROR, FINISHED
     }
 
-    RemoteReindexMigration start(RemoteReindexRequest request);
+    String start(RemoteReindexRequest request);
```

Comment:
On the interface level, I'd add a small javadoc to inform that returned String is supposed to be a migration ID.